### PR TITLE
ECL-498: Accessibility Issues - Data not maintained for Other Journey In Registration

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

**Review by liandra-giacomine — COMMENTED** [review]
Is it just the business name that we need to remember? When the user inputs the rest of the information is that being maintained aswell? If you tested and it all looks good I'm more than happy to approve, just thought it would be more complicated :D

**Comment by nemanjatomanovic** [comment]
> Is it just the business name that we need to remember? When the user inputs the rest of the information is that being maintained aswell? If you tested and it all looks good I'm more than happy to approve, just thought it would be more complicated :D

Business name is the only thing that is common across all of our Other entities, other information like CRN or CHRN are not present in all the entity types 😄

File: app/uk/gov/hmrc/economiccrimelevyregistration/cleanup/OtherEntityTypeDataCleanup.scala [hunk 1]
```diff
@@ -23,7 +23,7 @@ import javax.inject.Inject
 class OtherEntityTypeDataCleanup @Inject() () extends DataCleanup {
   def cleanup(registration: Registration): Registration = {
     val otherEntityJourneyData = registration.otherEntityJourneyData.copy(
-      businessName = None,
+      businessName = registration.otherEntityJourneyData.businessName,
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyregistration/cleanup/OtherEntityTypeDataCleanup.scala:26]: Do we need this line if we are not making changes? We are copying and overriding fields where necessary.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/OtherEntityTypeController.scala [hunk 2]
```diff
@@ -65,17 +65,35 @@ class OtherEntityTypeController @Inject() (
...
-            eclRegistrationConnector
-              .upsertRegistration(
-                dataCleanup.cleanup(
-                  request.registration.copy(
-                    optOtherEntityJourneyData = Some(otherEntityJourneyData)
+            request.registration.otherEntityJourneyData.entityType match {
+              case None                     =>
+                eclRegistrationConnector
+                  .upsertRegistration(
+                    request.registration.copy(
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/OtherEntityTypeController.scala:72]: to make it more readable can just pass the value into the call? 
e.g.
```
val updatedRegistration = request.registration.copy(optOtherEntityJourneyData = Some(otherEntityJourneyData))
eclRegistrationConnector.upsertRegistration(updatedRegistration)
```

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/OtherEntityTypeController.scala [hunk 3]
```diff
@@ -65,17 +65,35 @@ class OtherEntityTypeController @Inject() (
...
+                  .map { updatedRegistration =>
+                    Redirect(pageNavigator.nextPage(mode, updatedRegistration))
+                  }
+              case Some(previousEntityType) =>
+                if (previousEntityType == entityType) {
+                  Future.successful(Redirect(routes.OtherEntityCheckYourAnswersController.onPageLoad()))
+                } else {
+                  eclRegistrationConnector
+                    .upsertRegistration(
+                      dataCleanup.cleanup(
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/OtherEntityTypeController.scala:85]: same here

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala [hunk 4]
```diff
@@ -333,6 +335,20 @@ trait EclTestData {
       } yield ValidRegistrationWithRegistrationType(registration.copy(registrationType = Some(Initial)))
     }
 
+  implicit val arbValidRegistrationWithDifferentEntityTypes: Arbitrary[ValidRegistrationWithDifferentEntityTypes] =
```
Comment (liandra-giacomine) [inline test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala:338]: We don't need to make arbitraries here for all different types of scenarios, we could just use the copy function in the test itself?

**Review by TheDanielDoyle — APPROVED** [review]

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**